extensions/Worker.Extensions.CosmosDB/src/Config/CosmosDBBindingOptions.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using Azure.Core;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Functions.Worker.Extensions.CosmosDB;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    internal class CosmosDBBindingOptions
+    {
+        public string? ConnectionString { get; set; }
+
+        public string? AccountEndpoint { get; set; }
+
+        public TokenCredential? Credential { get; set; }
+
+        internal string BuildCacheKey(string connectionString, string region) => $"{connectionString}|{region}";
+        internal ConcurrentDictionary<string, CosmosClient> ClientCache { get; } = new ConcurrentDictionary<string, CosmosClient>();
+
+        internal virtual CosmosClient GetClient(string connection, string preferredLocations = "")
+        {
+            if (string.IsNullOrEmpty(connection))
+            {
+                throw new ArgumentNullException(nameof(connection));
+            }
+
+            string cacheKey = BuildCacheKey(connection, preferredLocations);
+
+            CosmosClientOptions cosmosClientOptions = new ()
```

Comment:
Customers have no way of configuring CosmosDB JSON serialization behavior. We should consider adding that (can be post-GA if we need).

Reply:
Would a good way to do this be to use the worker options serializer here?

Reply:
Might be possible. It won't be the same as Azure Track 2 SDKS, as they directly take `ObjectSerializer`. CosmosDB has a different serializer built into it, so would need to do whatever shimming necessary to integrate the worker options serializer there.

Reply:
Is this what you had in mind: https://github.com/Azure/azure-functions-dotnet-worker/pull/1779/files